spinalcordtoolbox/scripts/sct_create_mask.py

```diff
@@ -242,23 +242,14 @@ def create_mask(param):
     for iz in range(0, nz, 1):
         if iz in z_centerline_not_null:
             cx[iz], cy[iz] = ndimage.measurements.center_of_mass(np.array(data_centerline[:, :, iz]))
-    # create 2d masks
-    file_mask = 'data_mask'
+
+    # create 2d masks at each axial slice
+    im_out = zeros_like(im_data)
     for iz in range(nz):
-        if iz not in z_centerline_not_null:
-            # write an empty nifty volume
-            img = nibabel.Nifti1Image(data_centerline[:, :, iz], None, hdr)
-            nibabel.save(img, (file_mask + str(iz) + '.nii'))
-        else:
+        if iz in z_centerline_not_null:
             center = np.array([cx[iz], cy[iz]])
-            mask2d = create_mask2d(param, center, param.shape, param.size, im_data=im_data)
-            # Write NIFTI volumes
-            img = nibabel.Nifti1Image(mask2d, None, hdr)
-            nibabel.save(img, (file_mask + str(iz) + '.nii'))
-
-    fname_list = [file_mask + str(iz) + '.nii' for iz in range(nz)]
-    im_list = [Image(fname) for fname in fname_list]
-    im_out = concat_data(im_list, dim=2).save('mask_RPI.nii.gz')
```

Comment:
We have [new test failures](https://github.com/neuropoly/spinalcordtoolbox/pull/3270/checks?check_run_id=2102732546), so I ran the test for `sct_dmri_moco` twice (before/after the changes) and [compared the headers](https://www.diffchecker.com/fuyL4uJh) of two generated masks. To me, it looks like the old mask has had mismatched affine matrices this whole time. 

I think it's because the old method concatenated several single-slice images. Notably, `concat_data` reuses the header from the first image in the list. So, because the first image is single-slice, `pixdim3` gets set to `1` in the output mask.

This is yet another header issue (#2454, #3174, #3232, #3254). I'm starting to worry that header issues are actually a systemic problem in SCT. As far as I understand, we haven't really been validating any properties of the header, which makes it's easy for issues to slip by unnoticed.